connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -194,6 +200,13 @@ public Connector getConnector(String connType) {
         return instantiateConnector(connectorClass);
     }
 
+    public static Set<Class<? extends Connector>> getConnectorPlugins() {
+        Reflections reflections = new Reflections(new ConfigurationBuilder().setUrls(ClasspathHelper.forJavaClassPath()));
```

Comment:
Scanning for plugins is actually quite slow. Should we cache the results after the first lookup? I'm not sure we need to worry about anything like dynamically loaded classes, so I think that would be a safe way to avoid the cost of this scan for every request.